### PR TITLE
UCX: add memory registration

### PR DESCRIPTION
**Pull request by Sergei-Lebedev**

### CODE REVIEW

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI status: <b>FAIL</b> (click for details)</summary>

**Note:** the artefacts will be deleted after 09-Apr-2020

Artefact file | Status
-- | --
[configure_sharp_startup.log](https://gist.githubusercontent.com/mellanox-github/f57239ad7df153cfc63d338dc2df3d1d/raw/89268b51abea47a3cf0cee7c26d69864546756f3/configure_sharp_startup.log) | :x: FAIL
[build_nccl_rdma_sharp_plugins.log](https://gist.githubusercontent.com/mellanox-github/f57239ad7df153cfc63d338dc2df3d1d/raw/bac65e43a48fe56e2345948294b1c7a9b4f60799/build_nccl_rdma_sharp_plugins.log) | :heavy_check_mark: PASS

**Comment by mellanox-github** [comment]
Mellanox CI FAILed.

**Comment by artemry-mlnx** [comment]
bot:mlx:test

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI status: <b>PASS</b#  > (click for details)</summary>

**Note:** the artefacts will be deleted after 09-Apr-2020

Artefact file | Status
-- | --
[build_nccl_rdma_sharp_plugins.log](https://gist.githubusercontent.com/mellanox-github/e5115a09ea2cc3abeb08a0d5870b10c4/raw/76aea0adbc92413c9f1586ff830569f343bf1576/build_nccl_rdma_sharp_plugins.log) | :heavy_check_mark: PASS
[configure_sharp_startup.log](https://gist.githubusercontent.com/mellanox-github/e5115a09ea2cc3abeb08a0d5870b10c4/raw/a6672f55fddfe37873f81189597ef01b5bd91329/configure_sharp_startup.log) | :heavy_check_mark: PASS
[configure_sharp_stop.log](https://gist.githubusercontent.com/mellanox-github/e5115a09ea2cc3abeb08a0d5870b10c4/raw/08198ca62010b3722771d737a97ffbaeae0aee55/configure_sharp_stop.log) | :heavy_check_mark: PASS
[run_nccl_test.log](https://gist.githubusercontent.com/mellanox-github/e5115a09ea2cc3abeb08a0d5870b10c4/raw/a384476e26843938a85a791af444f7b8a98e380c/run_nccl_test.log) | :heavy_check_mark: PASS

**Comment by mellanox-github** [comment]
Mellanox CI PASSed.

File: src/ucx_plugin.c [hunk 1]
```diff
@@ -469,14 +469,54 @@ ncclResult_t nccl_ucx_accept(void *listen_comm, void **recv_comm) {
   return ncclSuccess;
 }
 
+#define REG_ALIGN (4096)
 ncclResult_t nccl_ucx_regmr(void* comm, void* data, int size, int type, void** mhandle) {
-  WARN("NET/UCX: not implemented");
-  return ncclInternalError;
+  ucp_mem_map_params_t mmap_params;
+  ucx_ctx_t            *ctx = (ucx_ctx_t*)comm;
```
Comment (bureddy) [inline src/ucx_plugin.c:475]: remove space and align names

File: src/ucx_plugin.c [hunk 2]
```diff
@@ -469,14 +469,54 @@ ncclResult_t nccl_ucx_accept(void *listen_comm, void **recv_comm) {
...
+  uint64_t             addr = (uint64_t)data;
+  ucx_mhandle_t *mh;
+  uint64_t      reg_addr, reg_size;
+  size_t        rkey_buf_size;
+  void          *rkey_buf;
+  
+  reg_addr = addr & (~(REG_ALIGN - 1));
+  reg_size = addr + size - reg_addr;
+  reg_size = ((reg_size + REG_ALIGN - 1) / REG_ALIGN ) * REG_ALIGN;
```
Comment (bureddy) [inline src/ucx_plugin.c:484]: use ROUNDUP macro?

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI status: <b>PASS</b#  > (click for details)</summary>

**Note:** the artefacts will be deleted after 10-Apr-2020

Artefact file | Status
-- | --
[build_nccl_rdma_sharp_plugins.log](https://gist.githubusercontent.com/mellanox-github/b86e1697645c66128836591fbfdfaf34/raw/300b36e962cef80c0c68ff1102bffc3b2e690aa3/build_nccl_rdma_sharp_plugins.log) | :heavy_check_mark: PASS
[configure_sharp_startup.log](https://gist.githubusercontent.com/mellanox-github/b86e1697645c66128836591fbfdfaf34/raw/bfb049106ddc5018ef9acdeb1337bc6085f98ca1/configure_sharp_startup.log) | :heavy_check_mark: PASS
[configure_sharp_stop.log](https://gist.githubusercontent.com/mellanox-github/b86e1697645c66128836591fbfdfaf34/raw/552deb2c7f9b7784c2aa6498fe345eda4f56f9f6/configure_sharp_stop.log) | :heavy_check_mark: PASS
[run_nccl_test.log](https://gist.githubusercontent.com/mellanox-github/b86e1697645c66128836591fbfdfaf34/raw/a5d67d66464f9e7b0ef6a6936d878a8450d7d118/run_nccl_test.log) | :heavy_check_mark: PASS

**Comment by mellanox-github** [comment]
Mellanox CI PASSed.

**Pull request merged**